### PR TITLE
updated modals

### PR DESCRIPTION
**Pull request by Philinphiladelphia**
## Issue Number

N/a

## Purpose/Implementation Notes

This combines all account creation modals into one configurable object. It also implements an option for non-dismissable modals.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

none

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: client/src/components/CreateOrLogin.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            href="https://orcid.org/"
+            label="Learn more at orcid.org"
+          />
+        </Box>
+      </Box>
+    </Box>
+  )
+}
+
+export const CreateOrLoginModal = ({ title, includeSignInSection }) => {
```
Comment (davidsmejia) [inline client/src/components/CreateOrLogin.js:121]: Could you call this `CreateOrLoginButton`

Comment (Philinphiladelphia) [reply client/src/components/CreateOrLogin.js:121]: Yep!

**Comment by davidsmejia** [comment]
Very nice. Just the one note about the Name

File: client/src/components/CreateOrLogin.js (hunk 1 above)
Comment (davidsmejia) [reply client/src/components/CreateOrLogin.js]: Sorry, could you name it `CreateAccountLoginButton` and then name the option `showSignIn` and set a default to the most common

Comment (davidsmejia) [reply client/src/components/CreateOrLogin.js]: `title` should also have a default that corresponds to the `showSignIn` default

**Comment by davidsmejia** [comment]
Could you also export `CreateOrLogin` and then update the filename to match the default export name too

File: client/src/components/Modal.js [hunk 2]
```diff
@@ -2,28 +2,32 @@ import Icon from 'components/Icon'
...
+    }
+  }
   return (
     <Box>
       {showing && (
-        <Layer
-          onEsc={() => setShowing(false)}
-          onClickOutside={() => setShowing(false)}
-        >
+        <Layer onEsc={dismissModal} onClickOutside={dismissModal}>
```
Comment (davidsmejia) [inline client/src/components/Modal.js:14]: should these be dependent on `nondismissable`

Comment (davidsmejia) [reply client/src/components/Modal.js:14]: nevermind that looks good

Comment (Philinphiladelphia) [reply client/src/components/Modal.js:14]: They are, as part of the dismissModal function.

**Pull request merged**